docs/config/01-overview.md

```diff
@@ -169,6 +169,111 @@ During production builds, the content of each generated file is hashed to repres
 hashFileNames: true
 ```
 
+## hydratedFlag
+
+Stencil has support for automatically applying a class or attribute to a
+component and all of its child components when they have finished hydrating.
+This can be used to prevent a [flash of unstyled content
+(FOUC)](https://en.wikipedia.org/wiki/Flash_of_unstyled_content), a
+typically-undesired 'flicker' of unstyled HTML that might otherwise occur
+during component rendering while various components are asynchronously
+downloaded and rendered.
+
+By default, Stencil will add the `hydrated` CSS class to elements to indicate
+hydration. The `hydratedFlag` config field allows this behavior to be
+customized, by changing the name of the applied CSS class, setting it to use an
+attribute to indicate hydration, or changing which type of CSS properties and
+values are assigned before and after hydrating. This config can also be used to
+turn off this behavior by setting it to `null`.
+
+If a Stencil configuration does not supply a value for `hydratedFlag` then
+Stencil will automatically generate the following default configuration:
```

Comment:
I based this on https://github.com/ionic-team/stencil/blob/4d9c73cc5c19d4cc7ff330e078cfddb063173724/src/declarations/stencil-public-compiler.ts#L210-L219